src/api/apiData.js

```diff
@@ -9,11 +9,16 @@
  */
 import columnWidths from '../selectors/columnWidths';
 import shallowEqualSelector from '../helper/shallowEqualSelector';
+import tableHeightsSelector from '../selectors/tableHeights';
 
 const getApiDataSelector = () =>
   shallowEqualSelector(
     [
       (state) => state.tableSize.height,
+      (state) =>
+        (state.headerHeight ? state.headerHeight : 0) +
+        (state.groupHeaderHeight ? state.groupHeaderHeight : 0) +
+        tableHeightsSelector(state).visibleRowsHeight,
```

Comment:
we dont need to calculate the value here since tableHeightsSelector already computes this value. You can use `componentHeight` prop returned by the tableHeightsSelector which also includes footer heights as we would also need to take that into account